compiler/rustc_passes/src/dead.rs

```diff
@@ -619,13 +618,8 @@ fn create_and_seed_worklist<'tcx>(
     // see `MarkSymbolVisitor::struct_constructors`
     let mut struct_constructors = Default::default();
     let mut worklist = access_levels
-        .map
         .iter()
-        .filter_map(
-            |(&id, &level)| {
-                if level >= privacy::AccessLevel::Reachable { Some(id) } else { None }
-            },
-        )
+        .filter_map(|(&id, _)| if access_levels.is_reachable(id) { Some(id) } else { None })
```

Comment:
This additional lookup in the map shouldn't be necessary, you can use something like
```suggestion
        .filter_map(|(&id, effective_vis)| if effective_vis.is_public_at_level(AccessLevel::Reachable) { Some(id) } else { None })
```
instead.

Reply:
Or
```suggestion
        .filter_map(|(&id, effective_vis)| effective_vis.is_public_at_level(AccessLevel::Reachable).then(id))
```